packages/page-staking/src/Nomination/useValidators.tsx

```diff
@@ -0,0 +1,57 @@
+import BN from 'bn.js';
+import { useEffect, useState } from 'react';
+import { useAccounts, useApi, useCall, useDebounce, useFavorites } from '@polkadot/react-hooks';
+import { DeriveStakingElected, DeriveSessionIndexes } from '@polkadot/api-derive/types';
+import { Balance } from '@polkadot/types/interfaces';
+import { SortBy, extractInfo, AllInfo, sort } from '@polkadot/app-staking/Targets';
+import { STORE_FAVS_BASE } from '@polkadot/app-staking/constants';
+import { Option } from '@polkadot/types';
+import { ValidatorInfo } from '@polkadot/app-staking/Targets/types';
+import useValidatorsFilter from './useValidatorsFilter';
+
+interface useValidatorsInterface {
+  validators: ValidatorInfo[];
+  validatorsLoading: boolean;
+}
+
+/**
+ * Get, sort and filter validators
+ * @return {Array<ValidatorInfo>} filtered validators
+ */
+function useValidators (): useValidatorsInterface {
```

Comment:
I need to see this in the editor to follow the flow here, this does indeed look like some copied code, so would like to understand the logic here. (So will get back to this in a next round, possibly only after some playing in the actual UI)